Modules/Forum/classes/class.ilForumTopic.php

```diff
@@ -565,40 +565,43 @@ public function movePosts($old_obj_id, $old_pk, $new_obj_id, $new_pk)
 				}
 			}
 
-			$this->db->lockTables(
-				array(
-					0 => array('name' => 'frm_user_read',     'type' => ilDBConstants::LOCK_WRITE),
-					1 => array('name' => 'frm_thread_access', 'type' => ilDBConstants::LOCK_WRITE)
-				)
-			);
+			$current_id = $this->id;
 
-			$this->db->manipulateF('
+			$ilAtomQuery = new ilAtomQuery($ilDB);
```

Comment:
Why not $ilDB->buildAtomQuery()? You could force the users of AtomQuery to code against an interface then, instead if requiring them to use a certain implementation. This will help one the AtomQuery gets more functionality which need to be implemented differently for different DBs.